src/word_debt_bot/cogs/game_commands.py

```diff
@@ -67,3 +67,17 @@ async def log(self, ctx, words: int):
             await ctx.send("Not registered! `.register`")
         except ValueError as err:
             await ctx.send(f"Error: {str(err)}")
+
+    @commands.command(name="leaderboard")
+    async def leaderboard(self, ctx, sort_by: str = "debt", lb_len: int = 5):
```

Comment:
Usability: I wonder if a page number would be better than a result count for displaying the leaderboard? If we specify a fixed size of each page, then we don't have to worry about exceeding the character limit (see below comment), and the user has more flexibility to go as far back in time as they want.

If we use [slice syntax](https://stackoverflow.com/a/509295/7543069) at `for u in users` we can easily get the range of users for a given page, without needing to use the `if i > lb_len` check.

Reply:
If we took the maximum discord username length, plus the length of the static string characters per leaderboard entry, plus the maximum length of a `WordDebtPlayer`'s `word_debt` & `cranes` to calculate the maximum possible size of a leaderboard entry and divide 2,000 by that number we could produce a figure to represent the maximum entries available for any given page of the leaderboard:

According to [Discord] (https://support.discord.com/hc/en-us/articles/12620128861463-New-Usernames-Display-Names) max username length is 32.

`max_entries = 2000 // (max_user_name_len<32> + static_chars +  leaderboard_values)`

`//` in Python produces whole numbers only if I remember correctly, but I would need to properly test this math.

I'm guessing it would yield 30ish entries per page max, but it depends on the maximum number of digits for `word_debt` and `cranes`.

#### Alternative

Alternatively, we could add an entry to a leaderboard page until the requested amount of entries has been met and each time the current page string cannot accept a new entry before going over 2000 we would add the current page to the a leaderboard list of page strings and start a new page (repeating the process each time it would overflow 2000), finally iterate through that leaderboard list sending each page until all have been sent. (Also, would need to test this solution)

`leaderboard()` command could contain something like:

```python
for page in lb:
    await ctx.send(page)
```

`create_leaderboard()` `for` loop could contain something like:

```python
if len(page + entry) > 2000:
    lb.append(page)
    page =""
page += entry
```

The downside would be multiple messages from the bot (bot name & icon etc.) separating the actual leaderboard output and may look cluttered or unsightly.

Do you have a preference?

### Added slicing syntax
Added slicing and tuple enumeration, now reads, `for u, i in users[:lb_len]` and removed the unnecessary `if i > lb_len` looks nice and looks to be working well.

Reply:
Could also simply limit leaderboard page size to 10.

Reply:
10 might be good just because even if we hit the theoretical limit it might flood the channel to send 30 at once, 10 is what other bots do and it feels visually nice + is a clean number.

Reply:
Agreed, its looks pretty clean.

#### Sample inputs & outputs

```
> .leaderboard
1. User 1 ...
...
10. User 10 ...
```

```
> .leaderboard debt 2 
11. User11 ...
...
20. User20 ...
```

(order by cranes)
```
> .leaderboard cranes
1. User42 ...
...
10. User33 ...
```

(end of leaderboard)
```
> .leaderboard debt 5
41. User41 ...
42. User42 ...
```

(Request page that does not exists)
```
> .leaderboard
The leaderboard is not that long! Last page = 5
```

#### Picture example

![Pic example](https://github.com/Swiddis/word-debt-bot/assets/134148713/8d994f10-6281-4e8a-ab2c-b12b189328a6)

#### Code Changes

Changed the name of variables and arguments to better reflect a leaderboard page and requested page.
Changed name of `create_leaderboard()` to `get_leaderboard_page()`.
Added a check for when a user requests a page that does not exists. 
Added arithmetic to get the range of leaderboard entries for the requested page.
Added a check to only print to the end of the leaderboard if last page does not have 10 entries.
Updated unit tests to reflect changes.

However, this solution requires `import math`.

For error handling of invalid input syntax such as `.leaderboard d ebt 2` I posted a possible solution in Issue #13.

Cheers!,
Greg